src/index.ts

```diff
@@ -27,6 +27,92 @@ function setupOpenMRS() {
       {
         id: "patient-chart-nav-items",
         load: getAsyncLifecycle(() => import("./nav.component"), options)
+      },
+      {
+        id: "summary-menu-item",
+        slot: "patient-chart-nav-menu",
+        load: getAsyncLifecycle(() => import("./menu-items/summary-link"), {
```

Comment:
@FlorianRappl , in Samuels code here, he's importing the link but this link is a function with is expecting the basePath. Once this code is moved to the widgets repo, how does it get this basePath? My understanding is that the repo doesn't actually know the basePath (nor should it)

Reply:
I think there is some confusion.

This is `summary-link`:

```jsx
export default function({ basePath }) {
  const { t } = useTranslation();

  return (
    <ConfigurableLink to={`${basePath}/summary`} className="bx--side-nav__link">
      {t("Summary", "Summary")}
    </ConfigurableLink>
  );
}
```

this is not a function (yes, from a JS POV, but continue reading...), but a **component**. `basePath` is a prop! This is a prop that is passed into the component (which is an extension) via the props.

This is what I mentioned. This is how its passed on.

Let me know if you follow this, otherwise I'm happy to have a more direct session with you on it.

Reply:
(Just for completeness)

Here is the `ExtensionSlot` that will ultimately render the `summary-link`:

```jsx
    <ExtensionSlot
      extensionSlotName="patient-chart-nav-menu"
      state={{ basePath, patientUuid }}
    />
```

As you can see `basePath` is passed in.

Reply:
It seems to me that this is resolved. Please re-open if not.